mlir/test/Dialect/Vector/canonicalize.mlir

```diff
@@ -1017,6 +1017,31 @@ func.func @canonicalize_broadcast_shapecast_to_shapecast(%arg0: vector<3x4xf32>)
 
 // -----
 
+
+// CHECK-LABEL: func @canonicalize_broadcast_shapecast_scalar
+//       CHECK:   vector.broadcast
+//  CHECK-SAME:   f32 to vector<3x4x1xf32>
+//   CHECK-NOT:   vector.shape_cast
+func.func @canonicalize_broadcast_shapecast_scalar(%arg0: f32) -> vector<3x4x1xf32> {
```

Comment:
[nit] How about more descriptive names: `@canonicalize_broadcast_shapecast_to_broadcast_scalar`. Similar comment for the other test.

Also, would you mind making sure tests for `ShapeCastBroadcastFolder` are clustered into two groups:
* shape_cast(broadcast) -> broadcast
* shape_cast(broadcast) -> shape_cast

? Thanks!